plume-web-jersey/src/main/java/com/coreoz/plume/jersey/security/control/ContentControlFeature.java

```diff
@@ -0,0 +1,168 @@
+package com.coreoz.plume.jersey.security.control;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.AnnotatedElement;
+
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.container.DynamicFeature;
+import jakarta.ws.rs.container.ResourceInfo;
+import jakarta.ws.rs.core.FeatureContext;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.ReaderInterceptor;
+import jakarta.ws.rs.ext.ReaderInterceptorContext;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ContentControlFeature implements DynamicFeature {
+
+    public static final int DEFAULT_MAX_SIZE = 500 * 1024; // 500 KB
+    private final Integer maxSize;
+
+    public ContentControlFeature(int maxSize) {
+        this.maxSize = maxSize;
+    }
+
+    public ContentControlFeature() {
+        this.maxSize = DEFAULT_MAX_SIZE;
+    }
+
+    public Integer getContentSizeLimit() {
+        if (maxSize == null) {
+            return DEFAULT_MAX_SIZE;
+        }
+        return maxSize;
+    }
+
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
+        addContentSizeFilter(resourceInfo.getResourceMethod(), context);
+    }
+
+    private void addContentSizeFilter(AnnotatedElement annotatedElement, FeatureContext methodResourcecontext) {
+        ContentSizeLimit contentSizeLimit = annotatedElement.getAnnotation(ContentSizeLimit.class);
+        methodResourcecontext.register(new ContentSizeLimitInterceptor(
+                contentSizeLimit != null ? contentSizeLimit.value() : maxSize
+            ));
+    }
+
+    public static class ContentSizeLimitInterceptor implements ReaderInterceptor {
+
+        private final int maxSize;
+
+        public ContentSizeLimitInterceptor(int maxSize) {
+            this.maxSize = maxSize;
+        }
+
+        // https://stackoverflow.com/questions/24516444/best-way-to-make-jersey-2-x-refuse-requests-with-incorrect-content-length
+        @Override
+        public Object aroundReadFrom(ReaderInterceptorContext context) throws IOException {
+            Integer headerContentLength;
+            try {
+                headerContentLength = Integer.parseInt(context.getHeaders().getFirst(HttpHeaders.CONTENT_LENGTH));
+            } catch (NumberFormatException e) {
+                headerContentLength = maxSize; // default value for GET or chunked body
+            }
```

Comment:
C'est un peu cracra, tu vas lever des exceptions à chaque requête GET.
Ce serait mieux de faire le parsing seulement si le header est non nul.

Par exemple :
```suggestion
                int headerContentLength = maxSize; // default value for GET or chunked body
                String contentLengthHeader = context.getHeaders().getFirst(HttpHeaders.CONTENT_LENGTH);
                if (contentLengthHeader) {
                  try {
                    headerContentLength = Integer.parseInt(contentLengthHeader);
                  } catch (NumberFormatException e) {
                    logger.warn("Wrong content length header received: {}", contentLengthHeader);
                  }
                }
```